packages/select/src/components/select/suggest.tsx

```diff
@@ -284,6 +285,23 @@ export class Suggest<T> extends React.PureComponent<ISuggestProps<T>, ISuggestSt
         Utils.safeInvokeMember(this.props.popoverProps, "onOpened", node);
     };
 
+    private handlePopoverClosed = (node: HTMLElement) => {
+        // The activeItem should always be the selectedItem when the Popover is first opened
+        // if the activeItem prop is not set. Set the activeItem on close so that there isn't
+        // a flash of the activeItem on screen.
+        const shouldResetActiveItemToSelectedItem =
+            this.props.activeItem === undefined && this.state.selectedItem && !this.props.resetOnSelect;
+
+        if (this.queryList && shouldResetActiveItemToSelectedItem) {
```

Comment:
nit: again, please avoid type coercion to boolean expressions... use `this.queryList != null && ...`